packages/apps/plugins/plugin-help/src/components/HelpContextProvider/HelpContextProvider.tsx

```diff
@@ -29,6 +29,39 @@ const removeTargetClass = (target: string | HTMLElement) => {
   }
 };
 
+const getTarget = (step: Step) => {
+  return typeof step.target === 'string' ? document.querySelector(step.target) : step.target;
+};
+
+/**
+ * Wait for the target element to be in the document.
+ */
+const waitForTarget = async (step: Step) => {
+  if (typeof step.target === 'string') {
+    const target = step.target;
+    const element = document.querySelector(target);
+    if (element) {
+      return;
+    }
+
+    await new Promise<void>((resolve) => {
+      const observer = new MutationObserver((mutations) => {
+        mutations.forEach((mutation) => {
+          if (mutation.addedNodes.length > 0) {
+            const element = document.querySelector(target);
+            if (element) {
+              observer.disconnect();
+              resolve();
+            }
+          }
+        });
+      });
+
+      observer.observe(document.body, { childList: true, subtree: true });
+    });
+  }
+};
```

Comment:
It might be nice to move this further upstream, it seems like it could be extremely useful, fine to start here of course.